From d6e3fc050f84f5b196511a800221cdf90a0e6fbf Mon Sep 17 00:00:00 2001
From: Pankaj Bansal <pankaj.bansal@nxp.com>
Date: Tue, 4 Dec 2018 01:16:52 +0530
Subject: [PATCH 601/706] can: flexcan: Fix error in setting fast propagation
 segment
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fast Propagation Segment Time = FPROPSEG × Time-Quanta.
This is different from all other timing parameters e.g
Fast Phase Segment 1 = (FPSEG1 + 1) × Time-Quanta.
Therefore, remove the "-1" when setting FPROPSEG.

Signed-off-by: Pankaj Bansal <pankaj.bansal@nxp.com>
(cherry picked from commit 96deaf87c778bbe8c81d37e5520e92684cfc39c7)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/can/flexcan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/can/flexcan.c b/drivers/net/can/flexcan.c
index b5607b312736..eeea5a665ffa 100644
--- a/drivers/net/can/flexcan.c
+++ b/drivers/net/can/flexcan.c
@@ -1087,7 +1087,7 @@ static void flexcan_set_bittiming(struct net_device *dev)
 
 		reg_fdcbt |= FLEXCAN_FDCBT_FPRESDIV(data_bt->brp - 1) |
 			     FLEXCAN_FDCBT_FRJW(data_bt->sjw - 1) |
-			     FLEXCAN_FDCBT_FPROPSEG(data_bt->prop_seg - 1) |
+			     FLEXCAN_FDCBT_FPROPSEG(data_bt->prop_seg) |
 			     FLEXCAN_FDCBT_FPSEG1(data_bt->phase_seg1 - 1) |
 			     FLEXCAN_FDCBT_FPSEG2(data_bt->phase_seg2 - 1);
 
-- 
2.17.1

